ios/UARCTModule/UARCTMessageCenter.m

```diff
@@ -24,22 +24,40 @@ + (UARCTMessageCenter *)shared {
     return sharedMessageCenterDelegate_;
 }
 
-#pragma mark UAInboxDelegate
+#pragma mark UAMessageCenterDisplayDelegate
 
 - (void)showMessageForID:(NSString *)messageID {
     if ([[NSUserDefaults standardUserDefaults] boolForKey:UARCTAutoLaunchMessageCenterKey]) {
-        [[UAirship messageCenter] displayMessageForID:messageID];
+        [[UAMessageCenter shared].defaultUI displayMessageCenterForMessageID:messageID animated:true];
     } else {
         [[UARCTEventEmitter shared] showInboxMessage:messageID];
     }
 }
 
-- (void)showInbox {
-    if ([[NSUserDefaults standardUserDefaults] boolForKey:UARCTAutoLaunchMessageCenterKey]) {
-        [[UAirship messageCenter] display];
-    } else {
-        [[UARCTEventEmitter shared] showInbox];
-    }
+- (void)displayMessageCenterAnimated:(BOOL)animated {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if ([[NSUserDefaults standardUserDefaults] boolForKey:UARCTAutoLaunchMessageCenterKey]) {
+            [[UAMessageCenter shared].defaultUI displayMessageCenterAnimated:animated];
+        } else {
+            [[UARCTEventEmitter shared] showInbox];
+        }
+     });
+
+}
+
+- (void)displayMessageCenterForMessageID:(NSString *)messageID animated:(BOOL)animated {
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Are these not running on the main queue?

Reply:
this is just cruft I pulled in when going off what we do in the samples, removed